### PR TITLE
Adding connectTimeout() and timeout() options

### PR DESCRIPTION
**Pull request by nexxai**
This PR adds support for the `connectTimeout()` (setting up the initial HTTP connection) and `timeout()` (the entire transaction) options.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,5 @@ yarn-error.log
 /.vscode
 composer.lock
 docker-compose.yml
```
Comment (jasonmccreary) [inline .gitignore]: This should not be necessary as it is no longer a Laravel app.

**Review by jasonmccreary — CHANGES_REQUESTED** [review]
Thanks. But I'm not sure why the `.gitignore` file was changed.

File: .gitignore (hunk 1 above)
Comment (nexxai) [reply .gitignore]: Oh, they must have been leftover files from a previous copy of the fork.  Apologies.  Will revert those specific changes.

**Comment by nexxai** [comment]
All done.  Ready for review.

**Review by jasonmccreary — APPROVED** [review]

**Comment by jasonmccreary** [comment]
Okay. I'm not sure why the build isn't running. But I'll go ahead and merge this as it should trigger on `main`.

**Pull request merged**